internal/tiltfile/tiltfile_state.go

```diff
@@ -1478,6 +1479,28 @@ func (s *tiltfileState) translateLocal() ([]model.Manifest, error) {
 
 	return result, nil
 }
+func (s *tiltfileState) warnDuplicateYamlResources(m model.Manifest) {
+	var tempName []string
+	deployTarget := m.K8sTarget()
+	displayNames := deployTarget.DisplayNames
+
+	duplicates := make(map[string]int)
+
+	// depends on the fact that the duplicate resource has formatting of doggos:service:default:core:n where n is nth
```

Comment:
good comment! I'd add a little more context: adding an index onto the end was a stopgap by us to keep Tilt from breaking when someone had the same entity twice in the same resource. We use it here to determine whether an entity appears more than once in this manifest. In future we may decide to make this an error instead (in which case we'd want to do this check closer to manifest assembly).

(The goal of a comment like this is to give the next dev to touch this code the tools they need to make decisions--so you're documenting not just WHAT you did, but WHY you did it, alternatives considered, etc.)